opennlp-tools/src/main/java/opennlp/tools/formats/irishsentencebank/IrishSentenceBankDocument.java

```diff
@@ -0,0 +1,262 @@
+            if (name.equals("flex")) {
+              String slottmp = sentnl.item(j).getAttributes().getNamedItem("slot").getNodeValue();
+              Integer slot = Integer.parseInt(slottmp);
+              if (slot > flexes) {
+                flexes = slot;
+              }
+
+              if (flx.get(slot) == null) {
+                flx.put(slot, new ArrayList<>());
+              }
+
+              String tkn = sentnl.item(j).getAttributes().getNamedItem("lemma").getNodeValue();
+              flx.get(slot).add(tkn);
+            } else if (name.equals("translation")) {
+              trans = sentnl.item(j).getFirstChild().getTextContent();
+            } else if (name.equals("original")) {
+              int last = 0;
+              NodeList orignl = sentnl.item(j).getChildNodes();
+              for (int k = 0; k < orignl.getLength(); k++) {
+                if (orignl.item(k).getNodeName().equals("token")) {
+                  String tmp = orignl.item(k).getFirstChild().getTextContent();
+                  spans.add(new Span(last, last + tmp.length()));
+
+                  String slottmp = orignl.item(k).getAttributes().getNamedItem("slot").getNodeValue();
+                  Integer slot = Integer.parseInt(slottmp);
+                  if (slot > flexes) {
+                    flexes = slot;
+                  }
+
+                  toks.put(slot, tmp);
+                  orig.append(tmp);
+                  last += tmp.length();              
+                } else if (orignl.item(k).getNodeName().equals("#text")) {
+                  String tmp = orignl.item(k).getTextContent();
+                  orig.append(tmp);
+
+                  if (!" ".equals(tmp)) {
+                    spans.add(new Span(advanceLeft(tmp, last), advanceRight(tmp, last)));
+                  }
+
+                  last += tmp.length();
+                } else {
+                  throw new IOException("Unexpected node: " + orignl.item(k).getNodeName());
+                }
+              }
+            } else if (name.equals("#text") || name.equals("#comment")) {
+              continue;
+            } else {
+              throw new IOException("Unexpected node: " + name);
+            }
+          }
+          IrishSentenceBankFlex[] flexa = new IrishSentenceBankFlex[flexes];
+          for (Integer flexidx : toks.keySet()) {
+            String left = toks.get(flexidx);
+            int rsize = flx.get(flexidx).size();
+            String[] right = new String[rsize];
+            right = flx.get(flexidx).toArray(right);
+            flexa[flexidx - 1] = new IrishSentenceBankFlex(left, right);
+          }
+
+          Span[] spanout = new Span[spans.size()];
+          spanout = spans.toArray(spanout);
+          document.add(new IrishSentenceBankSentence(src, trans, orig.toString(), spanout, flexa));
+        } else if (sentnode.getNodeName().equals("#text") || sentnode.getNodeName().equals("#comment")) {
+          continue;
```

Comment:
Can we avoid the 'naked continue' here, one way would be to invert the previous if condition